source/compiler/qsc_linter/src/lints/hir.rs

```diff
@@ -342,3 +343,38 @@ impl HirLintPass for DeprecatedDoubleColonOperator {
         }
     }
 }
+
+#[derive(Default)]
+struct AmbiguousUnaryOperatorAfterIf {
+    level: LintLevel,
+}
+
+/// Creates a lint for ambiguous unary operators after if-statements.
+/// For example:
+/// ```qsharp
+/// if condition { a } else { b } - c
+/// ```
+/// The user likely intended this to subtract `c` from the result of the if-expression,
+/// but to achieve this they need to wrap the if-expression in parentheses, like so:
+/// ```qsharp
+/// (if condition { a } else { b }) - c
+/// ```
+impl HirLintPass for AmbiguousUnaryOperatorAfterIf {
+    fn check_block(&mut self, block: &Block, buffer: &mut Vec<Lint>, _compilation: Compilation) {
+        if block.stmts.len() < 2 {
+            // Not enough statements to have an if-expression followed by a unary operator.
+            return;
+        }
+        for i in 0..(block.stmts.len() - 1) {
+            if let StmtKind::Expr(expr) = &block.stmts[i].kind
+                && matches!(&expr.kind, ExprKind::If(..))
+                && let StmtKind::Expr(next_expr) = &block.stmts[i + 1].kind
+                && matches!(next_expr.kind, ExprKind::UnOp(UnOp::Pos | UnOp::Neg, _))
+                && expr.ty != Ty::UNIT
+                && next_expr.ty != Ty::UNIT
```

Comment:
Is that possible? Users can't really have a `-()`.

Reply:
That's a good point, this can probably be removed as an extraneous check.